Program.cs

```diff
@@ -1,178 +1,264 @@
-                var _filePath = Path.Combine(_dirPath, $"{item.Hash}-{item.Width}x{item.Height}{Path.GetExtension(item.ImageURL)}");
-                var _filePathPart = $"{_filePath}.part";
-
-                if (File.Exists(_filePath)) {
-                    Globals.Logs.Add($"Skipping \u001b[22;37m{Path.GetFileName(_filePath)}\u001b[0m");
-                    continue;
-                }
-
-                Directory.CreateDirectory(_dirPath);
-
-                while (downloadThreads.Count >= maxThreads) { }
-
-                Globals.Logs.Add($"Downloading \u001b[22;37m{Path.GetFileName(_filePath)}\u001b[0m");
-
-                var _task = Task.Run(async () => {
-                    using var _dlstrm = await Globals.HttpClient.GetStreamAsync(item.ImageURL);
-                    using var _wstrm = File.Open(_filePathPart, FileMode.Create);
-
-                    _dlstrm.CopyTo(_wstrm);
-
-                    File.Move(_filePathPart, _filePath);
-                    Globals.Logs.Add($"Saved \u001b[22;37m{Path.GetFileName(_filePath)}\u001b[0m");
-                    progress++;
-                });
-
-                downloadThreads.Add(_task);
-                _ = Task.Run(() => {
-                    while (!_task.IsCompleted) {}
-
-                    lock (downloadThreadsLock) {
-                        downloadThreads.Remove(_task);
-                    }
-                });
-            }
-        }, booru, tags, downloadPath, cunnyAPIUrl, amount, skip, threads);
-
-        searchCommand.SetHandler(async (string booru, string tags, string? cunnyapiUrl, int amount, int skip, bool showTags, bool outputJson) => {
-            var results = await CunnyAPIClient.Get(cunnyapiUrl ?? Globals.DefaultCunnyAPIURL, booru, tags, amount, skip);
-
-            if (outputJson) {
-                Console.Write(JsonSerializer.Serialize(results));
-                return;
-            }
-
-            for (int i = 0; i < results.Count; i++) {
-                var item = results[i];
-
-                Console.ForegroundColor = i % 2 == 0 ? ConsoleColor.Cyan : ConsoleColor.Blue;
-                Console.WriteLine($"{i}: {(showTags ? $"[ {string.Join(' ', item.Tags)} ]" : "")}({item.Width}x{item.Height}) - {item.PostURL}");
-                Console.ResetColor();
-            }
-        }, booru, tags, cunnyAPIUrl, amount, skip, showTags, outputJson);
-
-        return await rootCommand.InvokeAsync(Args);
+RootCommand rootCommand = new("Uooh CLI. Can query images and download images")
+{
+    searchCommand,
+    downloadCommand
+};
+
+foreach (var option in new Option[]
+         {
+             booru,
+             tags,
+             amount,
+             threads,
+             showTags,
+             outputJson,
+             skip,
+             cunnyApiUrl,
+             excludeTags,
+             proxyOption,
+             downloadPath
+         })
+    rootCommand.AddOption(option);
+
+var progress = 0;

```

Comment:
Can you change this to 1?

```suggestion
var progress = 1;
```